### DISCUSSION TITLE
Raw prompt mode and function calling support

### DISCUSSION THREAD

**Discussion by kkolago-discovery** [amithkoujalgi/ollama4j#57] [Q&A] [answered]
Is it possible to send raw prompt to the models through Ollama4j? (I would like to use function calling provided by Mistral 7B v0.3 which require Ollama's raw mode in order to send special tokens in the prompt; see: [https://ollama.com/library/mistral](https://ollama.com/library/mistral) ).

Currently, I can see that JSONs send to Ollama have `raw` field set to `false`, and I don't know how to change it.

**Comment by amithkoujalgi** [comment]
@kkolago-discovery I have added support for function calling with tools but I am unable to publish the package at the moment due to technical issues with https://central.sonatype.org.
The [builds](https://github.com/amithkoujalgi/ollama4j/actions/runs/9907238679) are failing due to that. Therefore, I would recommend you to download the source and build the code to continue using the library until the package publishing issue is resolved.

Here's the link to the documentation for function calling with tools:
https://amithkoujalgi.github.io/ollama4j/docs/apis-generate/generate-with-tools

Cheers!

**Comment by amithkoujalgi** [comment, marked as answer]
Available in this release: https://github.com/amithkoujalgi/ollama4j/releases/tag/1.0.75

**Comment by kkolago-discovery** [comment]
Amazing. Thank you.

**Reply by amithkoujalgi** [reply]
Thanks! If your question has been answered, please do consider clicking on the `Mark as answer` button for my response. 🙏🏽
